src/app/features/search-bar/search-bar.component.ts

```diff
@@ -0,0 +1,250 @@
+import {
+  Component,
+  ElementRef,
+  EventEmitter,
+  Input,
+  OnDestroy,
+  Output,
+  ViewChild,
+  AfterViewInit,
+  ChangeDetectionStrategy,
+} from '@angular/core';
+import { T } from '../../t.const';
+import { FormControl } from '@angular/forms';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { tap, map, take, debounceTime, startWith } from 'rxjs/operators';
+import { TaskService } from '../tasks/task.service';
+import { Router } from '@angular/router';
+import { TODAY_TAG } from '../tag/tag.const';
+import { Project } from '../project/project.model';
+import { Tag } from '../tag/tag.model';
+import { ProjectService } from '../project/project.service';
+import { TagService } from '../tag/tag.service';
+import { Task } from '../tasks/task.model';
+import { blendInOutAnimation } from 'src/app/ui/animations/blend-in-out.ani';
+import { AnimationEvent } from '@angular/animations';
+import { SearchItem, SearchQueryParams } from './search-bar.model';
+import { getWorklogStr } from '../../util/get-work-log-str';
+
+@Component({
+  selector: 'search-bar',
+  templateUrl: './search-bar.component.html',
+  styleUrls: ['./search-bar.component.scss'],
+  animations: [blendInOutAnimation],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SearchBarComponent implements AfterViewInit, OnDestroy {
+  @Input() isElevated: boolean = false;
+  @Output() blurred: EventEmitter<any> = new EventEmitter();
+
+  @ViewChild('inputEl') inputEl!: ElementRef;
+  @ViewChild('searchForm') searchForm!: ElementRef;
+
+  T: typeof T = T;
+  isLoading$: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
+  taskSuggestionsCtrl: FormControl = new FormControl();
+  filteredIssueSuggestions$: Observable<SearchItem[]> = new Observable();
+  isArchivedTasks: boolean = false;
+  tooManyResults: boolean = false;
+
+  private _attachKeyDownHandlerTimeout?: number;
+  private _tasks: Task[] = [];
+  private _projects: Project[] = [];
+  private _tags: Tag[] = [];
+  private _searchableItems: SearchItem[] = [];
+
+  constructor(
+    private _taskService: TaskService,
+    private _projectService: ProjectService,
+    private _tagService: TagService,
+    private _router: Router,
+  ) {
+    this._taskService.allTasks$.pipe(take(1)).subscribe((t) => (this._tasks = t));
+    this._projectService.list$.pipe(take(1)).subscribe((p) => (this._projects = p));
+    this._tagService.tags$.pipe(take(1)).subscribe((t) => (this._tags = t));
+
+    this._mapTasksToSearchItems();
```

Comment:
I'd think so, too, but it works. this._tasks will be set inside the _mapTasksToSearchItems() method.
I don't really understand how that is possible. The assignment of _tasks should be waiting on the event queue at this point in time, and should therefore still be an empty array.
I could put the _mapTasksToSearchItems method in a later Angular lifecycle method but to be honest: I never really understood when the first tick on the event loop happens in-between/after those methods.

Reply:
> I could put the _mapTasksToSearchItems method in a later Angular lifecycle method but to be honest: I never really understood when the first tick on the event loop happens in-between/after those methods.

You can't count on this timing. What you can do is make _searchableItems to another observable that you assign via .pipe(map(()=>...))

Reply:
should be taken care of now. 
As a side note, this worked because subscription to store [will always deliver synchronously](https://github.com/ngrx/platform/issues/227#issuecomment-431682349). But, of course, this or the interface of this app might change, so better be safe on this one.

Reply:
That's pretty crazy :D I would have never thought that subscribe is executed right away for store values (or anything really), which makes it a bit confusing I'd say as it might be sometimes sync and async and sometimes async only. But well something rather important learned today! Thank you!

Reply:
Can you maybe add a comment about that above the subscriptions?

Reply:
Those subscriptions don't exist anymore in the latest version... If it comes up again, I'll add a comment. And yeah, pretty weird behavior...